src/CSharpExtDownloader.ts

```diff
@@ -9,34 +9,27 @@ import * as util from './common';
 import { Logger } from './logger';
 import { PackageManager, Status, PackageError } from './packages';
 import { PlatformInformation } from './platform';
+import { SetStatus, GetNetworkDependencies, ReportInstallationError, SendInstallationTelemetry } from './downloader.helper';
 
 /*
  * Class used to download the runtime dependencies of the C# Extension
  */
-export class CSharpExtDownloader
-{
+export class CSharpExtDownloader {
     public constructor(
         private channel: vscode.OutputChannel,
         private logger: Logger,
         private reporter: TelemetryReporter /* optional */,
         private packageJSON: any) {
     }
 
-    public installRuntimeDependencies(): Promise<boolean> {
+    public async installRuntimeDependencies(): Promise<boolean> {
         this.logger.append('Installing C# dependencies...');
+        this.logger.appendLine();
         this.channel.show();
 
-        let statusItem = vscode.window.createStatusBarItem(vscode.StatusBarAlignment.Right);
-        let status: Status = {
-            setMessage: text => {
-                statusItem.text = text;
-                statusItem.show();
-            },
-            setDetail: text => {
-                statusItem.tooltip = text;
-                statusItem.show();
-            }
-        };
+        let statusObject = SetStatus();
+        let status = statusObject.Status;
+        let statusItem = statusObject.StatusItem;
```

Comment:
nit: Instead of having `SetStatus` return the `StatusItem`, I think the code would be a lot more understandable if if you just returned `Status`, and `Status` had a `Dispose` method to dispose the `StatusItem`.